Monthly report January 2024

ch1bo opened the pull request:
TODO:
 - [x] @locallycompact on the API change
 - [x] @ch1bo conclusion
 - [x] Go through comments from technical writing

---

<!-- Consider each and tick it off one way or the other -->
* [x] CHANGELOG updated or not needed
* [x] Documentation updated or not needed
* [x] Haddocks updated or not needed
* [x] No new TODOs introduced or explained herafter

ffakenz committed:
Start monthly report for January 2024

v0d1ch committed:
Hydra chess section

ffakenz committed:
Add pitch about hydra-explorer development effort

ch1bo committed:
Work around two math rendering errors

No idea why parsing failed here / mathjax rendering was not used.
Working around the two usages as they are currently not critical.

Fixes #1261

ch1bo committed:
Write roadmap update

ch1bo committed:
Add hydra chess image and reword paragraphs

ch1bo committed:
Mark API changes section as TODO

ch1bo committed:
Write community update and draft conclusion

ch1bo committed:
Write a conclusion

locallycompact committed:
monthly: update JSON/CBOR section

ch1bo committed:
Apply suggestions from code review

Co-authored-by: Arnaud Bailly <79840582+abailly-iohk@users.noreply.github.com>

ch1bo committed:
Add a warning to the breaking API change section

ch1bo committed:
Reflow paragraphs

ch1bo committed:
Add dan as author

olgahryniuk committed:
Comments from technical writers

commit a4733754cfab60c6b7e3f7f29df7b8d06b0969fa
Author: olgahryniuk <67585499+olgahryniuk@users.noreply.github.com>
Date:   Wed Jan 31 14:00:37 2024 +0200

    Final TW review: several improvements

commit 05c117a22d7fb2d10650267f9c5395417eee8138
Merge: d1251a003 7cf5eb2b3
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Wed Jan 31 10:51:11 2024 +0000

    seventh review

commit d1251a00307a861bb5f59d1b9a2d429ed12e836c
Merge: e52467ee5 675a16427
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Wed Jan 31 10:35:21 2024 +0000

    sixth review

commit e52467ee59372ca68490c053ae49554c66a0d578
Merge: b9178ca60 d7adcc6d2
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Wed Jan 31 10:34:42 2024 +0000

    fifth review

commit b9178ca600b9125189f8eeb88acfae4953a1f420
Merge: 40eb0bb4d 684a3b7db
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Wed Jan 31 10:25:32 2024 +0000

    fourth review

commit 40eb0bb4d8f63a0e6c1a81dfdf2cd54545f99e05
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Tue Jan 30 18:19:47 2024 +0000

    third review

commit 684a3b7db2e0f6e220c810d154c8fc4b79fee0b2
Author: oduameh <emmanuel.ameh@iohk.io>
Date:   Tue Jan 30 18:19:47 2024 +0000

    hydra monthly dev report 2

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
 ![The roadmap with features and ideas](./img/2024-01-roadmap.jpg) <small><center>The latest roadmap with features and ideas</center></small>
 
-#### Notable updates
+### Notable updates
```
@olgahryniuk @oduameh The smaller font renders better on our docusaurus.

olgahryniuk replied:
Agreed, we followed the heading order rules, but that's not critical at all.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-* New feature to indicate preparation of [running latest version cardano-nodes in P2P
-#1256](https://github.com/input-output-hk/hydra/issues/1256).
-  This is necessary as the non-P2P relay nodes of IOG are to be shut down in January 2024.
+* Implemented a new feature, indicating preparation for [running the latest version of Cardano nodes in P2P #1256](https://github.com/input-output-hk/hydra/issues/1256). This is necessary as IOG's non-P2P relay nodes are to be shut down in January 2024.
```
Suggestion: revert this.

@olgahryniuk @oduameh This is not a report, but a list of notable updates _to the roadmap_. That means, that these features may also be plans and are not always a report on what happened. This item in particular was not done yet so why we only mention it's new.

olgahryniuk replied:
Thanks for the clarification. Suggest:
* A new feature indicating preparation for [running the latest version of cardano-nodes in P2P
#1256](https://github.com/input-output-hk/hydra/issues/1256).
  This is necessary as IOG's non-P2P relay nodes are to be shut down in January 2024.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-  #1257](https://github.com/input-output-hk/hydra/issues/1257) which will serve
-  as another proof point of being able to open/close heads in the new Conway
-  era.
+* Worked on implementing the new feature, [running smoke tests in SanchoNet #1257](https://github.com/input-output-hk/hydra/issues/1257), which will provide additional evidence of the capability to open and close heads in the Conway era.
```
Suggestion: revert this.

@olgahryniuk @oduameh Same as above.

olgahryniuk replied:
Agreed. Suggestion:
* Similarly, a new feature – [run smoke tests in SanchoNet
  #1257](https://github.com/input-output-hk/hydra/issues/1257) – will serve
  as another proof point of being able to open/close heads in the new Conway
  era.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-* Renamed "Drop Babbage support #1178" to [Switch L2 ledger to Conway
-  #1178](https://github.com/input-output-hk/hydra/issues/1178) to better capture
-  it's content.
+* Renamed 'Drop Babbage support #1178' to [Switch layer 2 ledger to Conway #1178](https://github.com/input-output-hk/hydra/issues/1178) to better capture its content.
```
@olgahryniuk @oduameh I see many line merges. Do you prefer markdown with fully connected paragraphs? It renders the same, but let's align on one format.

olgahryniuk replied:
I would so much appreciate connected lines and sentences. This would make working in suggestion mode much simpler.

ch1bo replied:
Fine by me, I'll connect all remaining ones.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-This representation is equivalent to the TextEnvelope produced by cardano-cli,
-except with an additional `txId` field, that is guaranteed to match the transaction
-id within the cbor object.
+This representation mirrors the TextEnvelope produced by the Cardano CLI, with the addition of an `txId` field, which is guaranteed to match the transaction ID within the CBOR object.
```
Suggestion: stick with `cardano-cli` to refer to the binary / tool

@olgahryniuk @oduameh

olgahryniuk replied:
Agreed 👍

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-are suboptimal or stumble accross a bug that needs fixing. Hydra Chess proved to
-be no different and we learned in the process of making this dApp.
+
+The team began developing a game on Hydra, leveraging dogfooding to uncover suboptimal elements or stumble upon bugs requiring resolution. Hydra Chess exemplified this process, offering valuable lessons during DApp development.
```
Suggestions: allow us to use the "we" form as we used this in the past

@olgahryniuk @oduameh The rewritten text here reads fine, but it can't shake off feeling weird when writing it (while it's very natural to just say "we")

olgahryniuk replied:
This thing relates to legal suggestions, but I understand :) Noted.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
 
-In this initial phase, we have implemented a basic backend service
+Measuring the progress and success of Hydra requires tools that provide
+insights into its usage. For this, the team initiated work on
```
@olgahryniuk @oduameh The first sentence was better before IMO (isn't this one lacking a verb?)

olgahryniuk replied:
I rewrote it to drop the use of 'we'. The first sentence was good otherwise.

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
-on Cardano.
+The team held the monthly review meeting for January 2024 on 2024-01-19 via Google Meet, presenting these [slides][slides] and this [recording][recording].
+
+Throughout the month, the teams demonstrated various Hydra and Mithril features. Additionally, Jose and his team from Modulo-Pi showcased a compelling demo on utilizing zero-knowledge cryptography in a Hydra head for a mastermind game.
```
@olgahryniuk @oduameh This is rewritten wrong. I wanted to refer to the review meeting this month.. not to the month.

olgahryniuk replied:
Agreed, let's revert

ch1bo commented on docs/monthly/2024-01-monthly.md:
```diff
@@ -12,150 +12,97 @@ developments to gather their feedback on proposed plans.
+
+Besides providing the usual status and roadmap updates on both Hydra and Mithril, several interested community members, including Tudor, joined the discussion to share their use case ideas. Tudor presented his proposal for supply chain tracking of Moldavian wine, which is currently running as a Catalyst proposal in Fund11. The proposal aims to leverage Hydra for CIP-68 NFT updates in a head, requ ...
+
+Charles provided feedback on considering Cardano block extensions as a potential core protocol change, enabling various proof systems and new consensus algorithms to support partner chains. The team emphasized collaborative efforts with the community to explore and develop the user space, determining which aspects could evolve into marketplaces and what should be integrated into core protocols. P ...
```
@olgahryniuk @oduameh 

"..., enabling various proof systems to be embedded and used by new consensus algorithms or better support for partner chains."

Also, I would like to keep the quotes around "user space" and "kernel space" as they are not (yet) standard terms in Cardano (we coined them last week with charles)

olgahryniuk replied:
Please use single quotes to adhere to the style guide.

ch1bo reviewed (commented):
@olgahryniuk @oduameh Commenting on the commit, unfortuantely without suggestions, but I feel this works much better than a separate PR which has conflicts.

Only some semantic changes needed and a few comments on aligning form between us. Please respond directly on the threads and I'll do the ammendments. Next time we can use the +- suggestions here on github or so.

ch1bo committed:
Addressed suggestions and comments

ch1bo commented:
@olgahryniuk Addressed all your suggestions and my corrections
@oduameh feel free to also give it a review (Files tab).

I think this is finished and we can publish by approving/merging.
